docs/pages/machine-id/guides/github-actions-kubernetes.mdx

```diff
@@ -0,0 +1,221 @@
+    # Teleport role configuration.
+    name: github-demo
+    apiGroup: rbac.authorization.k8s.io
+roleRef:
+  kind: ClusterRole
+  # "view" is a default ClusterRole that grants read-only access to resources
+  # See: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#user-facing-roles
+  name: view
+  apiGroup: rbac.authorization.k8s.io
+```
+
+Apply this to your cluster using `kubectl`:
+
+```
+$ kubectl apply -f ./clusterolebinding.yaml
+```
+
+## Step 4/4. Create the GitHub Actions workflow
+
+With all the configuration in Teleport and Kubernetes complete, you can now
+create a GitHub Actions workflow that can connect to your Kubernetes cluster.
+
+Our example workflow will list all of the pods contained within the cluster,
+but this could equally be modified to deploy to a Kubernetes cluster with
+`kubectl` or `helm`.
+
+In the GitHub workflows directory of your repository (`.github/workflows/`)
+create a new workflow YAML file, in this case `actionstest.yml`.
+
+Insert the following, ensuring that you replace `my-kubernetes-cluster` with the
+name of your Kubernetes cluster and `example.domain:443` with the address of
+your Teleport Proxy.
+
+```yaml
+# This is a basic workflow to help you get started, modify it for your needs.
+on:
+  push:
+    branches:
+      - main
+jobs:
+  demo:
+    permissions:
+      # The "id-token: write" permission is required or Machine ID will not be 
+      # able to authenticate with the cluster.
+      id-token: write
+      contents: read
+    name: guide-demo
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Fetch kubectl
+        uses: azure/setup-kubectl@v3
+      - name: Fetch Teleport binaries
+        uses: teleport-actions/setup@v1
+        with:
+          version: (=teleport.version=)
+      - name: Fetch credentials using Machine ID
+        uses: teleport-actions/auth-k8s@v1
+        with:
+          # Use the address of the auth/proxy server for your own cluster.
+          proxy: example.domain:443
+          # Use the name of the join token resource you created in step 1.
+          token: github-token
+          # Use the name of your Kubernetes cluster
+          kubernetes-cluster: my-kubernetes-cluster
+      - name: List pods
+        run: kubectl get pods -A
+```
+
+The `auth-k8s` action sets the `KUBECONFIG` for future steps to the credentials
+it has fetched from Teleport and this means that most existing tooling for
+Kubernetes (e.g `kubectl` and `helm`) will be configured to use your cluster
+with no additional configuration.
+
+Add, commit, and push this new workflow file to the `main` branch of your
+repository.
+
+Navigate to the **Actions** tab of your GitHub repository in your web browser.
+Select the **Workflow** that has now been created and triggered by the change,
+and select the `guide-demo` job.
+
+Expand the List pods step of the action, where you can then confirm that the
```

Comment:
If "List" is capitalized because "List pods" is the name of a button or tool, please identify it as such. If it's a button, make it bold. If it's a link, wrap in quotes.